src/settings.js

```diff
@@ -2,44 +2,76 @@ import { isPlainObject } from './helpers';
 import { Value } from './values';
 
 export default class Settings {
-  static prefix;
-
-  constructor(settings) {
-    this.cache = new Map();
-    this.proxy = new Proxy(settings, {
-      get: (target, key) => {
-        if (this.cache.has(key)) {
-          return this.cache.get(key);
-        }
+  static assign(...args) {
```

Comment:
I think I understand this, but `assign` is a name that confuses me. I guess it comes from `Object.assign`? To me this would make more sense as something like `Settings.merge`, since settings objects are immutable. Also a docstring would be nice.

Reply:
Yeah i just pulled `assign` from `Object`. But `merge` makes sense too.

I need to do a full pass over all the code and add docstrings.